crates/bootstrap_srv/src/test.rs

```diff
@@ -0,0 +1,748 @@
+}
+
+#[test]
+fn reject_get_no_space() {
+    let s = BootstrapSrv::new(Config::testing()).unwrap();
+
+    let addr = format!("http://{:?}/bootstrap", s.listen_addr());
+    match ureq::get(&addr).call() {
+        Err(ureq::Error::Status(_status, err)) => {
+            let err = err.into_string().unwrap();
+
+            println!("{err:?}");
+
+            assert!(err.to_string().contains("InvalidPathSegment"));
+        }
+        oth => panic!("unexpected {oth:?}"),
+    }
+}
+
+#[test]
+fn reject_put_no_space() {
+    let s = BootstrapSrv::new(Config::testing()).unwrap();
+
+    let addr = format!("http://{:?}/bootstrap", s.listen_addr());
+    match ureq::put(&addr).call() {
+        Err(ureq::Error::Status(_status, err)) => {
+            let err = err.into_string().unwrap();
+
+            println!("{err:?}");
+
+            assert!(err.to_string().contains("InvalidPathSegment"));
+        }
+        oth => panic!("unexpected {oth:?}"),
+    }
+}
+
+#[test]
+fn reject_put_no_agent() {
+    let s = BootstrapSrv::new(Config::testing()).unwrap();
+
+    let addr = format!("http://{:?}/bootstrap/{}", s.listen_addr(), S1);
+    match ureq::put(&addr).call() {
+        Err(ureq::Error::Status(_status, err)) => {
+            let err = err.into_string().unwrap();
+
+            println!("{err:?}");
+
+            assert!(err.to_string().contains("InvalidPathSegment"));
+        }
+        oth => panic!("unexpected {oth:?}"),
+    }
+}
+
+#[test]
+fn reject_mismatch_agent_url() {
+    let s = BootstrapSrv::new(Config::testing()).unwrap();
+
+    let err = PutInfo {
+        addr: s.listen_addr(),
+        agent_url: Some("AAAA"),
+        ..Default::default()
+    }
+    .call()
+    .unwrap_err();
+
+    assert!(err.to_string().contains("InvalidAgent"));
+}
+
+#[test]
+fn reject_mismatch_space_url() {
+    let s = BootstrapSrv::new(Config::testing()).unwrap();
+
+    let err = PutInfo {
+        addr: s.listen_addr(),
+        space_url: "AAAA",
+        ..Default::default()
+    }
+    .call()
+    .unwrap_err();
+
+    assert!(err.to_string().contains("InvalidSpace"));
+}
+
+#[test]
+fn reject_old_created_at() {
+    let s = BootstrapSrv::new(Config::testing()).unwrap();
+
+    let err = PutInfo {
+        addr: s.listen_addr(),
+        created_at: 0,
+        ..Default::default()
+    }
+    .call()
+    .unwrap_err();
+
+    assert!(err.to_string().contains("InvalidCreatedAt"));
+}
+
+#[test]
+fn reject_future_created_at() {
+    let s = BootstrapSrv::new(Config::testing()).unwrap();
+
+    let err = PutInfo {
+        addr: s.listen_addr(),
+        created_at: i64::MAX - 500,
+        expires_at: i64::MAX,
+        ..Default::default()
+    }
+    .call()
+    .unwrap_err();
+
+    assert!(err.to_string().contains("InvalidCreatedAt"));
+}
+
+#[test]
+fn reject_expired() {
+    let s = BootstrapSrv::new(Config::testing()).unwrap();
+
+    let expires_at = crate::now() - 500;
+    let created_at = crate::now() - 1500;
```

Comment:
created_at can be valid here, right?

```suggestion
    let created_at = crate::now();
```

Reply:
I don't want to trigger the other failure case of rejecting puts where the expired_at <= created_at, so I need the created_at here to be before the expired_at here.